### PR TITLE
Std futures and async/await

### PR DESCRIPTION
**Pull request by usbalbin**
I have updated to use std futures, tokio 0.2 and async/await.

I also replaced uses of `uninitialized()` with `[0; SIZE];` and removed `Bytes` dependency, please tell me if I should undo any of that.

### CODE REVIEW

File: src/async.rs [hunk 1]
```diff
@@ -5,49 +5,48 @@ use std::{
...
     timeout: Option<Duration>,
 }
 
 impl AsyncRequest {
-    fn send_and_recv(self, pdu: pdu::Buf) -> SnmpFuture<Bytes> {
-        let fut = self
-            .socket
-            .send_dgram(Bytes::from(&pdu[..]), &self.destination)
-            .map_err(|_| SnmpError::SendError);
+    async fn send_and_recv(mut self, pdu: pdu::Buf) -> SnmpResult<Vec<u8>> {
```
Comment (ancwrd1) [inline src/async.rs]: I am wondering if consuming `self` here is still necessary with async/await.

File: src/async.rs [hunk 2]
```diff
@@ -171,7 +166,7 @@ impl AsyncSession {
     ///   - `Timeticks`
     ///   - `Opaque`
     ///   - `Counter64`
-    pub fn set(mut self, values: &[(&[u32], Value)]) -> SnmpFuture<SnmpPdu> {
+    pub async fn set(mut self, values: &[(&[u32], Value)]) -> SnmpResult<SnmpPdu> {
```
Comment (ancwrd1) [inline src/async.rs]: Again, may be we don't need `self` by value here?

File: src/async.rs (hunk 1 above)
Comment (usbalbin) [reply src/async.rs]: Oh right, thanks :)

Comment (usbalbin) [reply src/async.rs]: Would it be useful to change `AsyncSession::req_id` to be AtomicI32 so all methods could take self by shared reference? I have hade some issues with the borrow checker with `.await` in closures with a mutable reference to `AsyncSession`.

Comment (ancwrd1) [reply src/async.rs]: If `mut` is only needed for `ref_id` then yes, I think it's better to change it.

**Pull request merged**